src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -508,7 +516,15 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
 
 #define UCT_DC_MLX5_CHECK_RES(_iface, _ep) \
     { \
-        ucs_status_t status = uct_dc_mlx5_iface_dci_get(_iface, _ep); \
+        ucs_status_t status = uct_dc_mlx5_iface_dci_get(_iface, _ep, 1); \
```

Comment:
so it means that now all AM functions except flow control will have extra overhead for checking iface resources?

Reply:
yes we will check for iface resources (rdma reads)

Reply:
why do we have to add a check for rdma reads and mpool_is_empty in am_short flow?

Reply:
to preserve ordering

Reply:
but it adds extra overhead
and we don't really need to preserve ordering between rdma_read and am's. for example we don't check FC for rdma_read.